lib/modules/platform/azure/index.ts

```diff
@@ -427,6 +417,21 @@ export async function getBranchStatus(
   return 'green';
 }
 
+async function getMergeStrategy(
+  targetRefName: string
+): Promise<GitPullRequestMergeStrategy> {
+  // TODO #7154
+  return (
+    config.mergeMethods[targetRefName] ??
+    (config.mergeMethods[targetRefName] = await azureHelper.getMergeMethod(
+      config.repoId,
+      config.project,
+      targetRefName,
+      config.defaultBranch
```

Comment:
what if base branches are in use? is this the target branch to merge against? we probably need to pass the base branch here.

Reply:
`targetRefName` is the target of the already-created PR.  By the time this is called, Renovate is either updating a PR it just created to enable platform autocomplete, or it's merging it.

This unifies the logic in both cases - previously when enabling platform autocomplete it was possible to select an invalid strategy depending on the order the DevOps API returns its policy list.

Reply:
but why do we pass the default branch? for what it is used?

Reply:
There is (apparently, I've never seen one in the wild) a way to make a DevOps policy targeting the default branch of a repo instead of a prefix or exact ref, and my read of `isRelevantScope` in azure-helper suggests that it needs to be passed the default branch name of that repo in order to correctly match against such a policy: https://github.com/renovatebot/renovate/blob/7f4c8ebeb919091b4773c177fa30f96873c6ac95/lib/modules/platform/azure/azure-helper.ts#L123-L141

Reply:
I've worked out how to create a default branch policy - they apply to the default branch of every repository in a project, and have `null` as their `refName`.  An example from the API response on my test repo:
```json
"settings": {
    "allowSquash": true,
    "allowRebase": true,
    "allowRebaseMerge": true,
    "scope": [
        {
            "refName": null,
            "matchKind": "DefaultBranch",
            "repositoryId": null
        }
    ]
},
```
So the function in azure-helper.ts does need to know the default branch name of the repository being worked on in order to check if there is a relevant policy.